code-challenges.test.js

```diff
@@ -21,30 +20,120 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+            [arr[currentIndex], arr[randomIndex]] = [arr[randomIndex], arr[currentIndex]];
+        }
+        return arr
+    }
+    shuffle(rm_First_Element)
+}
+
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the net total of votes.
 
+// PSEUDO CODE
+// create a function that takes an object of key/value vote: up or down
+// return total votes
+
+
 // a) Create a test with expect statements for each of the variables provided.
 
 const votes1 = {upVotes: 13, downVotes: 2}
 // Expected output: 11
 const votes2 = {upVotes: 2, downVotes: 33}
 // Expected output: -31
 
+describe('voteCounter', () => {
+    it('takes in an object that contains upvotes and downvotes and returns the net total of votes', () => {
+        expect(voteCounter(votes1)).toEqual(11);
+        expect(voteCounter(votes2)).toEqual(-31);
+    })
+})
+
+// GOOD FAIL:
+// vote_counter › takes in an object that contains upvotes and downvotes and returns the net total of votes
+// ReferenceError: voteCounter is not defined
+
 
 // b) Create the function that makes the test pass.
 
+const voteCounter = (obj) => {
+    let voteSum = (obj.upVotes - obj.downVotes)
+    return voteSum};
+
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
+// PSEUDO CODE
+// create a function that takes in two arrays
+// iterate over each array
+// if value doesn't exist in third array, add to third array
+// push the value into a third array
+// return the array with NO DUPLICATE VALUES
+// STRETCH: No thanks.
+
+
 // a) Create a test with an expect statement using the variables provided.
 
 const dataArray1 = ["array", "object", "number", "string", "Boolean"]
 const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
+describe("twoArraysNoDuplicates", () => {
+    it("Create a function that takes in two arrays as arguments and returns one array with no duplicate values", () => {
+        expect(twoArraysNoDuplicates(dataArray1, dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"]);
+    })
+})
+
+// GOOD FAIL
+// twoArraysNoDuplicates › Create a function that takes in two arrays as arguments and returns one array with no duplicate values
+// ReferenceError: twoArraysNoDuplicates is not defined
 
 // b) Create the function that makes the test pass.
+
+const twoArraysNoDuplicates = (arr1, arr2) => {
+    let noDuplicates = [];
+    arr1.map((value) => {
+       if (noDuplicates.includes(value) === false) {
+        noDuplicates.push(value)
+       } 
+    })
+    arr2.map((value) => {
+        if (noDuplicates.includes(value) === false) {
+         noDuplicates.push(value)
+        } 
+     })
+    return noDuplicates
+}
```

Comment:
Unique approach. Did this function pass? If so, please refactor. Go back and see if there is way that you can practice the DRY principle.  May be there is a way to combine the two arrays and then return an array with the values that true to a condition that is set? You can do all of this without needing an empty array to store the values.

Reply:
Yes, the function passed the test.